### PR TITLE
docs: README audit table

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/blobstream-contracts/issues/250

### CODE REVIEW

**Review by rach-id — APPROVED** [review]
Thanks for helping with this,

**Comment by rootulp** [comment]
np! Let's hold off on merging this because Informal is going to provide an updated PDF with a fixed commit hash link.

**Comment by rootulp** [comment]
@rach-id thoughts on merging this as is? We haven't received a response from Binary Builders

**Comment by rach-id** [comment]
It was this: https://github.com/celestiaorg/blobstream-contracts/releases/tag/v2.1.2

File: README.md [hunk 1]
```diff
@@ -93,3 +93,11 @@ Relayed tuples are in the same order as Celestia block headers.
...
+|------------|-----------------------------------------------|-------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------|
+| 2023/10/17 | [Binary Builders](https://binary.builders/)   | [v1.0.0-rc10](https://github.com/celestiaorg/celestia-app/releases/tag/v1.0.0-rc10) | [eb7a4e7](https://github.com/celestiaorg/blobstream-contracts/commit/eb7a4e74718b80277ad9dde116ead67383f5fe15) | [binary-builders.pdf](https://github.com/celestiaorg/blobstream-contracts/files/13961809/2023-10-17_Celestia_Audit_Report_Binary_Builders.pdf) |
```
Comment (rootulp) [inline README.md:101]: The commit hash for blobstream-contracts wasn't included in the Binary Builders PDF but Frojdi from Binary Builders shared this one.

**Comment by rootulp** [comment]
Ready for review @rach-id @adlerjohn

File: README.md [hunk 2]
```diff
@@ -93,3 +93,11 @@
...
+| 2023/10/26 | [Informal Systems](https://informal.systems/) | [v1.0.0](https://github.com/celestiaorg/celestia-app/tree/v1.0.0)                   | [cf301adf](https://github.com/celestiaorg/blobstream-contracts/blob/cf301adfbfdae138526199fab805822400dcfd5d)  | [informal-systems.pdf](https://github.com/celestiaorg/blobstream-contracts/files/13961767/Celestia_.Q4.2023.QGB-v2-20231026_182304.pdf)        |
+| 2023/11/16 | [Ottersec](https://osec.io/)                  | [v1.3.0](https://github.com/celestiaorg/celestia-app/releases/tag/v1.3.0)           | [v3.1.0](https://github.com/celestiaorg/blobstream-contracts/releases/tag/v3.1.0)                              | [ottersec.pdf](https://github.com/celestiaorg/blobstream-contracts/files/14383577/celestia_blobstream_audit_final.pdf)                         |
```
[the remark this answers is not in the document]
Comment (rach-id) [reply README.md:103]: @rootulp not blocking but might be nice to commit this

**Review by rach-id — APPROVED** [review]

**Pull request merged**